pkg/driverbuilder/builder/image.go

```diff
@@ -45,78 +63,110 @@ func (im ImagesMap) findImage(target Type, gccVers semver.Version) (Image, bool)
 	return Image{}, false
 }
 
-func (b *Build) LoadImages() {
+func (f *FileImagesLister) LoadImages(build *Build) {
+	// loop over lines in file to print them
+	file, err := os.Open(f.FilePath)
+	if err != nil {
+		logger.WithError(err).WithField("FilePath", f.FilePath).Fatal("error opening builder repo file")
+	}
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		infos := strings.Split(scanner.Text(), ",")
+		if len(infos) < 3 {
+			logger.WithField("FilePath", f.FilePath).WithField("line", scanner.Text()).Fatal("Invalid image list file: expected at least 3 fields (name,target,gcc_version) but got " + strconv.Itoa(len(infos)) + ".")
+		}
+		name := infos[0]
+		target := Type(infos[1])
+		gccVersions := infos[2:]
+		for _, gcc := range gccVersions {
+			buildImage := Image{
+				Name:       name,
+				Target:     target,
+				GCCVersion: mustParseTolerant(gcc),
+			}
+			build.Images[buildImage.toKey()] = buildImage
```

Comment:
I'd just make this method return a slice of Images, it is a much better API (and is unit testable); then main `image.ListImages` will gather all the images and fill the build.Images map.

Reply:
Found a way to eliminate ref to builder inside objects and function of ImageLister